app/api/helpers/permission_manager.py

```diff
@@ -129,6 +130,11 @@ def is_coorganizer_or_user_itself(view, view_args, view_kwargs, *args, **kwargs)
     if user.is_organizer(kwargs['event_id']) or user.is_coorganizer(kwargs['event_id']):
         return view(*view_args, **view_kwargs)
 
+    if ('model' in kwargs) and (kwargs['model'].__name__ == 'Speaker'):
+        query_user = Speaker.query.filter_by(email=user._email).first()
+        if query_user:
+            return view(*view_args, **view_kwargs)
```

Comment:
Create a separate annotation

This level of meta programming is not good

Reply:
@iamareebjamal Are you referring to the usage of `kwargs['model'].__name__ == 'Speaker'`?

Reply:
Yes

Reply:
@iamareebjamal I have updated it. Please have a look?